### PR TITLE
Use Buildkite for CI

### PR DESCRIPTION
**Pull request by turt2live**
TODO: Better pipelines (see https://github.com/vector-im/riot-web/issues/9019)

Part of fixing https://github.com/vector-im/riot-web/issues/9164

**Partnered with**:
* https://github.com/matrix-org/matrix-js-sdk/pull/859
* https://github.com/matrix-org/matrix-react-sdk/pull/2788

### CODE REVIEW

**Comment by turt2live** [comment]
This PR doesn't have status checks yet because for some reason BuildKite isn't sending them (or can't?)

File: scripts/fetch-develop.deps.sh [hunk 1]
```diff
@@ -40,9 +40,8 @@ function dodep() {
     # Try the PR author's branch in case it exists on the deps as well.
     # Try the target branch of the push or PR.
     # Use the default branch as the last resort.
-    if [[ "$TRAVIS" == true ]]; then
-        clone $org $repo $TRAVIS_PULL_REQUEST_BRANCH ||
```
Comment (jryans) [inline scripts/fetch-develop.deps.sh]: It looks like `$BUILDKITE_BRANCH` is the PR author's branch, while `$BUILDKITE_PULL_REQUEST_BASE_BRANCH` is the target branch. I think we can use both to preserve current behaviour here.

**Comment by turt2live** [comment]
> Nit: Buildkite seems to spell their name with lowercase "K".

Can I just disagree with their branding and continue using bad habits? 

(I'll go around fixing it)

File: scripts/fetch-develop.deps.sh (hunk 1 above)
Comment (turt2live) [reply scripts/fetch-develop.deps.sh]: Whups, good catch. Didn't mean to drop this.

**Pull request merged**